src/alchemlyb/preprocessing/subsampling.py

```diff
@@ -97,14 +98,16 @@ def statistical_inefficiency(df, series=None, lower=None, upper=None, step=None)
 
         # calculate statistical inefficiency of series
         statinef  = statisticalInefficiency(series)
-
-        # we round up
-        statinef = int(np.rint(statinef))
-
-        # subsample according to statistical inefficiency
-        series = series.iloc[::statinef]
-
-        df = df.loc[series.index]
+        
+        #use the subsampleCorrelatedData function to get the subsample index
+        indices = subsampleCorrelatedData(series, g=statinef)
```

Comment:
Can you briefly explain what `subsampleCorrelatedData` does (and maybe link to the code in their repo)?

Reply:
@orbeckst yes, the link to the code is https://github.com/choderalab/pymbar/blob/ac648e63787592a251992a764876713a64f17455/pymbar/timeseries.py#L632 it basically subsample the time series (`series`) based on the given `g` (statistical inefficiency)

Reply:
I don't quite see the point of this change. From what I can tell the `subsampleCorrelatedData` function [deals in integer-spaced samples, too](https://github.com/choderalab/pymbar/blob/ac648e63787592a251992a764876713a64f17455/pymbar/timeseries.py#L715). @shuail can you explain how this approach really differs from the one we already have here?

Reply:
@dotsdl yes, the code you refer to is an optional option `conservative`, and the default for that option is `False`. I didn't notice that option before while looking at the details, I think the implementation is also different from what alchemlyb currently had. 

The topic here is basically all related to the precision and distinguishability of the subsampling. For a given `g` like 1.5, current alchemlyb code will round up using `int(np.rint(1.5))` = 1.0 and, the `conservative` option will round to 2 using `math.ceil(1.5)` and the `non conservative` option will use 1.5. For a given time series with 1,000 data point, current alchemlyb; `conservative`; `non conservative`  will get 1,000; 500; 666 data point respectively. Since the energy will depend on the subsampling, I think we should be careful about how we round the `g` and I suggested to use the `non conservative` option here to make all dataset as much distinguishable as possible.

Reply:
Add explicit keywords for `subsampleCorrelatedData(..., fast=False, conservative=False, verbose=False)` (especially `conservative=False` to make sure that the behavior in alchemlyb won't change, even if pymbar changes its defaults.)

Reply:
I agree with https://github.com/alchemistry/alchemlyb/pull/38#discussion_r150987109 – until we come up with anything better inside alchemlyb I would also go with the pymbar version.

Reply:
@orbeckst yes, sounds good to me. I will add the default options explicitly.